utils/network.go

```diff
@@ -18,9 +18,9 @@ type HostPort struct {
 	Port    int32
 }
 
-func TcpCheck(ip string, port int32) bool {
+func TcpCheck(hp *HostPort) bool {
 	timeout := 5 * time.Second
-	conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%d", ip, port), timeout)
+	conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%d", hp.Address, hp.Port), timeout)
```

Comment:
can `conn != nil` even if `err !=nil`? if not one could use:
```
        conn, _ := net.DialTimeout("tcp", fmt.Sprintf("%s:%d", hp.Address, hp.Port), timeout)
	if conn != nil {
		return true
	}
	return false
```

Reply:
conn cannot be non-nil when err is non-nil, so this works out every time.

Reply:
I always have troubles with double negations :smile: error non-nil means conn = nil, right? So the code here can be simplified?
Can't see that this is explained here either https://golang.org/pkg/net/#Dial.

Reply:
yes I think that should work. Ill simplify the condition. Thanks @mgfeller